featureflags/evaluations/evaluator.py

```diff
@@ -0,0 +1,303 @@
+            if self._is_enabled(target, distribution.bucket_by,
+                                total_percentage):
+                log.debug("Enabled for distribution %s", distribution)
+                return variation
+        log.debug("Variation of distribution evaluation %s", variation)
+        return variation
+
+    def _check_target_in_segment(self, segments: List[str],
+                                 target: Target) -> bool:
+        for segment_identifier in segments:
+            segment = self.provider.get_segment(segment_identifier)
+
+            if segment:
+                # Should Target be excluded - if in excluded
+                # list we return false
+                if not isinstance(segment.excluded, Unset) and \
+                        target.identifier in segment.excluded:
+                    log.debug('Target % s excluded from segment % s' +
+                              'via exclude list\n',
+                              target.name, segment.name)
+                    return False
+
+                # Should Target be included - if in included list
+                #  we return true
+                if not isinstance(segment.included, Unset) and \
+                        target.identifier in segment.included:
+                    log.debug('Target %s included in segment %s' +
+                              ' via include list\n',
+                              target.name, segment.name)
+                    return True
+
+                # Should Target be included via segment rules
+                if segment.rules and self._evaluate_clauses(segment.rules,
+                                                            target):
+                    log.debug('Target %s included in segment %s via rules\n',
+                              target.name, segment.name)
+                    return True
+        log.debug("Target groups empty return false")
+        return False
+
+    def _evaluate_clause(self, clause: Clause, target: Target) -> bool:
+        if clause.op is None:
+            log.debug("Clause is empty")
+            return False
+
+        if not clause.values:
+            log.debug("Clause values is empty")
+            return False
+
+        operator = clause.op.lower()
+        type = target.get_type(clause.attribute)
+
+        if type is None:
+            if operator == SEGMENT_MATCH_OPERATOR.lower():
+                log.debug("Clause operator is %s, evaluate on segment",
+                          operator)
+                return self._check_target_in_segment(clause.values, target)
+            log.debug("Attribute type %s is none return false", type)
+            return False
+        log.debug("evaluate clause with object %s operator %s and value %s",
+                  object, operator, clause.values)
+        if operator == IN_OPERATOR.lower():
+            return type.in_list(clause.values)
+        if operator == EQUAL_OPERATOR.lower():
```

Comment:
This is a nitpick on syntax, since the behavior shouldn't change at all. But the subsequent `if` in this block should be `elif`

Reply:
OK, but when I use `if` and `return` I never use `elif`, if you prefer this I can change. I don't use elif often